ArcGISRuntimeSDKQt_CppSamples/Routing/FindClosestFacilityToMultipleIncidentsService/FindClosestFacilityToMultipleIncidentsService.cpp

```diff
@@ -0,0 +1,241 @@
+  // apply the facilities symbol as the renderer for this layer
+  m_facilitiesFeatureLayer->setRenderer(new SimpleRenderer(m_facilitySymbol, this));
+
+  // create service feature table from feature server
+  m_incidentsFeatureTable = new ServiceFeatureTable(QUrl("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/San_Diego_Incidents/FeatureServer/0"), this);
+  // create feature layer from feature service table
+  m_incidentsFeatureLayer = new FeatureLayer(m_incidentsFeatureTable, this);
+  // apply the incidents symbol as the renderer for this layer
+  m_incidentsFeatureLayer->setRenderer(new SimpleRenderer(m_incidentSymbol, this));
+
+  // connect to the doneLoading signal which calls the slot to set the view point geometry
+  connect(m_facilitiesFeatureTable, &ServiceFeatureTable::doneLoading, this, &FindClosestFacilityToMultipleIncidentsService::setViewpointGeometry);
+  connect(m_incidentsFeatureTable, &ServiceFeatureTable::doneLoading, this, &FindClosestFacilityToMultipleIncidentsService::setViewpointGeometry);
+
+  // load both service feature tables
+  m_facilitiesFeatureTable->load();
+  m_incidentsFeatureTable->load();
+
+}
+
+void FindClosestFacilityToMultipleIncidentsService::setupRouting()
+{
+  // connect to createDefaultParametersCompleted signal of the closest facility task
+  connect(m_task, &ClosestFacilityTask::createDefaultParametersCompleted, this, [this]
+          (QUuid, Esri::ArcGISRuntime::ClosestFacilityParameters defaultParameters)
+  {
+    // create query parameters to query features from feature table
+    QueryParameters params;
+    params.setWhereClause("1=1");
+    m_facilityParams = defaultParameters;
+    // set closest facilities parameters facilities
+    m_facilityParams.setFacilitiesWithFeatureTable(m_facilitiesFeatureTable, params);
+    // set closest facilitites parameters incidents
+    m_facilityParams.setIncidentsWithFeatureTable(m_incidentsFeatureTable, params);
+
+    // disable busy indicator
+    m_busy = false;
+    // enable solve button
+    m_solveButton = true;
+    emit busyChanged();
+    emit solveButtonChanged();
+  });
+
+  // connect to solveClosestFacilityCompleted signal of the closest facility task
+  connect(m_task, &ClosestFacilityTask::solveClosestFacilityCompleted, this, [this]
+          (QUuid, ClosestFacilityResult closestFacilityResult)
+  {
+    if(closestFacilityResult.isEmpty())
+    {
+      qDebug() << "Empty result";
+      return;
+    }
+
+    // loop through all incidents
+    for(int incidentIndex = 0; incidentIndex < m_incidentsFeatureTable->numberOfFeatures(); incidentIndex++)
+    {
+      // get the index of the facility closest to the incident
+      int closestFacilityIndex = closestFacilityResult.rankedFacilityIndexes(incidentIndex).first();
+      // get the route between the incident and the facility
+      ClosestFacilityRoute route = closestFacilityResult.route(closestFacilityIndex,incidentIndex);
```

Comment:
formatting: `closestFacilityIndex, incidentIndex`

Also, `ClosestFacilityRoute` can be const.